elide-datastore/elide-datastore-inmemorydb/src/main/java/com/yahoo/elide/datastores/inmemory/InMemoryTransaction.java

```diff
@@ -162,32 +163,20 @@ public Object loadObject(Class<?> entityClass,
         if (objs == null) {
             return Collections.emptyList();
         }
+        // Support for filtering
+        if (filterExpression.isPresent()) {
+            Set<Predicate> predicates = filterExpression.get().accept(new PredicateExtractionVisitor());
```

Comment:
I don't think this is quite what you want.  A predicate extraction visitor would collapse an expression like
`(A or B) and not C)` to a set of predicates `[A,B,C]` which in turn the code below treats like `(A and B and C)`.